sympy/functions/special/zeta_functions.py

```diff
@@ -214,6 +214,16 @@ def _eval_rewrite_as_zeta(self, z, s, a, **kwargs):
     def _eval_rewrite_as_polylog(self, z, s, a, **kwargs):
         return self._eval_rewrite_helper(polylog)
 
+    def _eval_is_finite(self):
+        z, s, a = self.args
+        if z == 0 or a == 0:
+            return z.is_zero and a.is_zero and s.is_nonpositive
```

Comment:
What should happen here if any of the `is_` checks gives `None`?

https://docs.sympy.org/latest/guides/booleans.html

Reply:
Hi @oscarbenjamin , thanks for taking a look and the information about the booleans.
I thought python and was compatible with the fuzzy logic but it is indeed not. I will use the fuzzy_and function, this should handle correctly the None cases.
Also, I am not sure what is the convention when the function evaluates to NaN, should the is_finite be None?

Reply:
> what is the convention when the function evaluates to NaN, should the is_finite be None?

Probably. What case evaluates to nan that you are thinking of?

Reply:
It is because a lot of situations where the function is infinite is because one has $0^{-s}$. The previous check `s.is_nonpositive` gives False when s is not real. I have added a check for s to be real, when s is complex the expression is undefined so now the function should return None.